catris/games/game_base_class.py

```diff
@@ -351,49 +359,53 @@ def get_square_texts(self, player: Player) -> dict[tuple[int, int], bytes]:
     def get_lines_to_render(self, rendering_for_this_player: Player) -> list[bytes]:
         pass
 
-    async def _explode_bombs(self, bombs: list[BombSquare]) -> list[BombSquare]:
+    async def _explode_bombs(self, bombs: set[tuple[int, int]]) -> set[tuple[int, int]]:
         exploding_points = {
             (x, y)
             for x, y in self.valid_landed_coordinates
-            for bomb in bombs
-            if (x - bomb.x) ** 2 + (y - bomb.y) ** 2 < 3.5**2
+            for bomb_x, bomb_y in bombs
+            if (x - bomb_x) ** 2 + (y - bomb_y) ** 2 < 3.5**2
         }
-        explode_next = [
-            square
-            for square in self._get_all_squares()
+        explode_next = {
+            point
+            for point, square in self._get_all_squares().items()
             if isinstance(square, BombSquare)
-            and (square.x, square.y) in exploding_points
-            and square not in bombs
-        ]
+            and point in exploding_points
+            and point not in bombs
+        }
 
         if exploding_points:
             await self.flash(exploding_points, 41)
-            for square in self.landed_squares.copy():
-                if (square.x, square.y) in exploding_points:
-                    self.landed_squares.remove(square)
+            self.landed_squares = {
+                point: square
+                for point, square in self.landed_squares.items()
+                if point not in exploding_points
+            }
```

Comment:
Should make a method that deletes squares everywhere. It's done in a couple places